audio.cpp

```diff
@@ -9,7 +9,12 @@ namespace Audio {
 	// ● check_audio
 	//-------------------------------------------------------------------------
 	inline SDL_AudioDeviceID check_audio(lua_State* L, int index) {
+		#if __x86_64__
+		long long id = (uintptr_t) Util::check_usertable(L, index, "audio device");
```

Comment:
Why are these intermediate casts required?

Reply:
Because on linux with both GCC&clang, the C compiler is confusing the exact bit length of the uintptr. Maybe it's Cygwin has different 64bit setup, or maybe it's something lying around header files. But if there is no such cast, the compiler will report an error of loss of information if 64bit value is cast into 32bit. This get rid of the error, while it still coredumps on Linux. So there are other issues with pointers here.

Reply:
I guess the reason why I haven't got this error message so far is that I've always been targeting 32-bit Win32. I've never heard of such annoyingly complicated pointers on x86_64!

My toolchain is MinGW with GCC 7.1.0 and I'm too lazy to get it updated, as configuring compilers and libraries on Windoge is just a plain pain. Hopefully that's enough for my use. Oh yup, and you know that Win32 really loved playing with packed structs, integer atoms stored in a pointer and `LOWORD` and `HIWORD` macros.

MS provides [specific functions dedicated to convert between pointers and integers](https://docs.microsoft.com/zh-cn/windows/desktop/WinProg64/the-tools#helper-functions) to deal with the x64 problem, while I don't think that strategy fits my small 'dumping project. So the next priority will be to refactor the ugly pointer part and instead try … a union? (Hey, that would be *much* uglier…) I have no idea in my mind yet.